apps/photos/src/components/PhotoList/index.tsx

```diff
@@ -265,6 +269,10 @@ export function PhotoList({
     const shouldRefresh = useRef(false);
     const listRef = useRef(null);
 
+    const [checkedDates, setCheckedDates] = useState({});
```

Comment:
checkedDates is an object to maintain select all checkbox state with its element being trimmed as the keys used in it are names of checkboxes.
These trimmed names will create complexity if we derive selectedDates from it which is a string array.

Reply:
why can't we directly use the `selectedDate` as the key of the checkbox, they are strings too, and will be unique, right?

Reply:
Yes you are right but then how will we maintain state of dates which were changed due to manual selection. If we only keep checkedDates object then we will lose manually selected files on a day (suppose Today) where select all is unchecked as when it will enter function onChangeSelectAllCheckBox if we click select all checkbox for some other day suppose yesterday then all manually selected files of today will get unchecked.